ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+  }
+
+  func showActivityIndicator() {
+    if !overlayWindow.isHidden && overlayWindow.viewWithTag(activityIndicatorViewTag) != nil {
+      return
+    }
+
+    dismissGetStartedView(nil)
+    overlayWindow.isHidden = false
+    let activityView = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
+    let containerView = UIView(frame: activityView.bounds.insetBy(dx: -10.0, dy: -10.0))
+    containerView.backgroundColor = UIColor(white: 0.5, alpha: 0.8)
+    containerView.layer.cornerRadius = 6.0
+    containerView.center = overlayWindow.center
+    containerView.tag = activityIndicatorViewTag
+    containerView.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin, .flexibleTopMargin,
+                                      .flexibleBottomMargin]
+    activityView.center = CGPoint(x: containerView.bounds.size.width * 0.5, y: containerView.bounds.size.height * 0.5)
+    activityView.startAnimating()
+    containerView.addSubview(activityView)
+    overlayWindow.addSubview(containerView)
+  }
+
+  func dismissActivityIndicator() {
+    overlayWindow.viewWithTag(activityIndicatorViewTag)?.removeFromSuperview()
+    overlayWindow.isHidden = true
+  }
+
+  func sliderValueChanged(_ sender: UISlider) {
+    textSize = CGFloat(sender.value + Float(minTextSize)).rounded()
+    textView.font = textView.font?.withSize(textSize)
+    let textSizeTitle = presentedViewController?.view.viewWithTag(2) as? UILabel
+    textSizeTitle?.text = "Text size: \(Int(textSize))"
+  }
+
+  func rectForBarButtonItem(_ barButtonItem: UIBarButtonItem) -> CGRect {
+    let view =  barButtonItem.value(forKey: "view") as? UIView
+    return view?.frame ?? CGRect.zero
+  }
+
+  func startTimer() {
+    if loadTimer == nil {
+      loadTimer = Timer.scheduledTimer(timeInterval: 1.0, target: self,
+                                       selector: #selector(self.timerAction), userInfo: nil, repeats: true)
+    }
+  }
+
+  func stopTimer() {
+    if let timer = loadTimer {
+      timer.invalidate()
+      loadTimer = nil
+    }
+  }
+
+  func timerAction() {
+    if updateStatus == 1 {
+      stopTimer()
+
+      dismissActivityIndicator()
+
+      textView.isEditable = true
+      textView.becomeFirstResponder()
+
+      if let launchUrl = launchUrl {
+        performAction(from: launchUrl)
+      } else {
+        loadSavedUserText()
+        if shouldShowGetStarted() {
+          perform(#selector(self.showGetStartedView), with: nil, afterDelay: 1.0)
+        }
+      }
+    } else if updateStatus == -1 {
+      stopTimer()
+      dismissActivityIndicator()
+      loadSavedUserText()
+      textView.isEditable = true
+      textView.becomeFirstResponder()
+      if shouldShowGetStarted() {
+        perform(#selector(self.showGetStartedView), with: nil, afterDelay: 1.0)
+      }
+    }
+  }
+
+  func resetTextViewCursor() {
+    textView.selectedRange = NSRange(location: 0, length: 0)
+
+    // TODO: Figure out what this does
```

Comment:
Swift is saying "size might not actually have a value, but if it does, then..."

Check to ensure that it will, since we might should log an error message otherwise.  Likewise for the following method.

Also, maybe pull the `// TODO` once that's done.

Reply:
That comment was referring to figuring out how increasing the font size and decreasing it helps in resetting the text view cursor.